### PR TITLE
feat(duplicated_node_checker): add duplicated node names to msg

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
### <samp>🤖 Generated by Copilot at 232b433</samp>

This pull request enhances the `duplicated_node_checker` node to optionally report the names of duplicated nodes in the diagnostic message. It also updates the configuration files and comments for the node and its usage in the `system_error_monitor` node.

---

add flag to show duplicated node names to msg (default false)
the motivation for this is it's necessary to check duplicated node name in the log file sometimes (e.g. running scenario simulator in CI/CD environment)

launcher [PR](https://github.com/autowarefoundation/autoware_launch/pull/651) should be merged first
 

## Tests performed

Run scenario_simulator_v2 in local environment.
I confirmed node name is added to msg when duplicated node is launched shown below.
```
[system_error_monitor-16] [ERROR 1698054734.955753195] [system_error_monitor /autoware/system/duplicated_node_checker/duplicated_node_checker: duplicated_node_checker]: [Single Point Fault]: Error: Duplicated nodes detected /simulation/openscenario_interpreter (logThrottledNamed():54)
```

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kyoichi-sugahara** [comment]
@Owen-Liuyuxuan @shmpwk 
Could you review this PR when you have time?

File: system/system_error_monitor/config/system_error_monitor.planning_simulation.param.yaml [hunk 1]
```diff
@@ -40,6 +40,7 @@
         /autoware/system/service_log_checker: { sf_at: "warn", lf_at: "none", spf_at: "none" }
         /autoware/system/duplicated_node_checker: default
         # /autoware/system/resource_monitoring: { sf_at: "warn", lf_at: "error", spf_at: "none" }
+        # /autoware/system/duplicated_node_checker: default
```
Comment (Owen-Liuyuxuan) [inline system/system_error_monitor/config/system_error_monitor.planning_simulation.param.yaml:43]: What is the reason for adding this commented line in this file?

Comment (kyoichi-sugahara) [reply system/system_error_monitor/config/system_error_monitor.planning_simulation.param.yaml:43]: It's to make it consistent with the config file on the launcher side, but the change is not necessary.
file:
https://github.com/autowarefoundation/autoware_launch/blob/13b28551b50141bd887042063e00cfc987c19e76/autoware_launch/config/system/system_error_monitor/system_error_monitor.planning_simulation.param.yaml#L42
FYI:
changed  by following PR:
https://github.com/autowarefoundation/autoware_launch/pull/649

**Review by shmpwk — APPROVED** [review]
LGTM

**Review by ito-san — APPROVED** [review]

**Pull request merged**